plugins/GSdx/Window/GSwxDialog.cpp

```diff
@@ -0,0 +1,917 @@
+    rend_hacks_grid->Add(mem_wrap_check);
+    rend_hacks_grid->Add(preload_gs_check);
+
+    upscale_hacks_grid->Add(align_sprite_check);
+    upscale_hacks_grid->Add(merge_sprite_check);
+    upscale_hacks_grid->Add(wild_arms_check);
+
+    auto *rend_hack_choice_grid = new wxFlexGridSizer(2, 0, 0);
+    auto *upscale_hack_choice_grid = new wxFlexGridSizer(2, 0, 0);
+
+    // Renderer Hacks:
+    add_label_and_combo_box(this, rend_hack_choice_grid, m_half_select, "Half Screen Fix:", theApp.m_gs_generic_list, IDC_HALF_SCREEN_TS);
+    add_label_and_combo_box(this, rend_hack_choice_grid, m_tri_select, "Trilinear Filtering:", theApp.m_gs_trifilter, IDC_TRI_FILTER);
+
+    // Skipdraw Range
+    add_label(this, rend_hack_choice_grid, "Skipdraw Range:");
+    auto *skip_box = new wxBoxSizer(wxHORIZONTAL);
+    skip_x_spin = new wxSpinCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxSP_ARROW_KEYS, 0, 10000, 0);
+    skip_y_spin = new wxSpinCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxSP_ARROW_KEYS, 0, 10000, 0);
+    add_tooltip(skip_x_spin, IDC_TCOFFSETX);
+    add_tooltip(skip_y_spin, IDC_TCOFFSETY);
+    skip_box->Add(skip_x_spin);
+    skip_box->Add(skip_y_spin);
+
+    rend_hack_choice_grid->Add(skip_box);
+
+    // Upscale Hacks:
+    add_label_and_combo_box(this, upscale_hack_choice_grid, m_gs_offset_hack_select, "Half-Pixel Offset:", theApp.m_gs_offset_hack, IDC_OFFSETHACK);
+    add_label_and_combo_box(this, upscale_hack_choice_grid, m_round_hack_select, "Round Sprite:", theApp.m_gs_hack, IDC_ROUND_SPRITE);
+
+    // Texture Offsets
+    add_label(this, upscale_hack_choice_grid, "Texture Offsets:");
+    auto *tex_off_box = new wxBoxSizer(wxHORIZONTAL);
+    tex_off_x_spin = new wxSpinCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxSP_ARROW_KEYS, 0, 10000, 0);
+    tex_off_y_spin = new wxSpinCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxSP_ARROW_KEYS, 0, 10000, 0);
+    tex_off_box->Add(tex_off_x_spin);
+    tex_off_box->Add(tex_off_y_spin);
+
+    upscale_hack_choice_grid->Add(tex_off_box);
+
+    rend_hacks_box->Add(rend_hacks_grid);
+    rend_hacks_box->Add(new wxStaticLine(this, wxID_ANY), wxSizerFlags().Centre().Expand().Border(5));
+    rend_hacks_box->Add(rend_hack_choice_grid, wxSizerFlags().Expand());
+
+    upscale_hacks_box->Add(upscale_hacks_grid);
+    upscale_hacks_box->Add(new wxStaticLine(this, wxID_ANY), wxSizerFlags().Centre().Expand().Border(5));
+    upscale_hacks_box->Add(upscale_hack_choice_grid, wxSizerFlags().Expand());
+
+    tab_box->Add(hacks_check, wxSizerFlags().Left());
+    tab_box->Add(rend_hacks_box, wxSizerFlags().Centre().Expand());
+    tab_box->Add(upscale_hacks_box, wxSizerFlags().Centre().Expand());
+
+    SetSizerAndFit(tab_box);
+    Bind(wxEVT_SPINCTRL, &HacksTab::CallUpdate, this);
+    Bind(wxEVT_CHECKBOX, &HacksTab::CallUpdate, this);
+}
+
+void HacksTab::CallUpdate(wxCommandEvent &event)
```

Comment:
You can do something like
```cpp
void HacksTab::CallUpdate(wxCommandEvent& /*event*/)
{
}
```

It'll save you a compiler warning. Same goes for anywhere else this occurs. It's a minor thing though so I'll leave it up to you.

Reply:
Done.